src/webui/prefjson.cpp

```diff
@@ -322,8 +322,12 @@ void prefjson::setPreferences(const QString& json)
         session->setGlobalDownloadSpeedLimit(m["dl_limit"].toInt());
     if (m.contains("up_limit"))
         session->setGlobalUploadSpeedLimit(m["up_limit"].toInt());
-    if (m.contains("enable_utp"))
-        session->setUTPEnabled(m["enable_utp"].toBool());
+    if (m.contains("bittorrent_protocol")) {
+        const int value = m["bittorrent_protocol"].toInt();
+        if ((static_cast<int>(BitTorrent::BTProtocol::Both) <= value)
+                && (value <= static_cast<int>(BitTorrent::BTProtocol::UTP)))
+            session->setBTProtocol(static_cast<BitTorrent::BTProtocol>(value));
```

Comment:
Maybe sanitation should be done inside setBTProtocol()?

Reply:
Instead of doing this each time we call setBTProtocol (and possibly forget to do it) we must check/normalize params inside it.

Reply:
Alright, done.